### PR TITLE
Display appropriate error when time between attempts condition is not satisfied

### LINKED ISSUE #459: Display appropriate error when time between attempts is not satisfied

### PR DESCRIPTION
**Pull request by ankitjavalkar**
Closes #459 

yaksh_attempt_error

### CODE REVIEW

File: yaksh/models.py [hunk 1]
```diff
@@ -1216,11 +1216,18 @@ def can_attempt_now(self, user, course_id):
             )
             if last_attempt:
                 time_lag = (timezone.now() - last_attempt.start_time).total_seconds() / 3600
-                return time_lag >= self.quiz.time_between_attempts
+                can_attempt = time_lag >= self.quiz.time_between_attempts
```
Comment (maheshgudi) [inline yaksh/models.py]: Instead this thing can be checked in views. This way this code remains intact.

File: yaksh/models.py [hunk 2]
```diff
@@ -1216,11 +1216,18 @@ def can_attempt_now(self, user, course_id):
...
+                msg = "You cannot start the next attempt for this quiz before {0} hour(s)".format(
+                    self.quiz.time_between_attempts
+                ) if not can_attempt else None
+                return can_attempt, msg
             else:
-                return True
+                return True, None
```
Comment (maheshgudi) [inline yaksh/models.py]: This is why doing it in views make sense. Sending None is pointless

Comment (ankitjavalkar) [reply yaksh/models.py]: I don't think that it is pointless, ```None``` just means that there is no error message. Pushing this to the views will mean adding if statements in a place that is already crowded with a lot of decision making in the first place (lines 543 to line 559 or so).

Returning all the necessary data from the models keeps the views clean.

File: yaksh/models.py (hunk 1 above)
Comment (ankitjavalkar) [reply yaksh/models.py]: I just changed the return values, earlier it used to be just a True/False now the error message also gets set here.

File: yaksh/views.py [hunk 3]
```diff
@@ -532,9 +532,10 @@ def start(request, questionpaper_id=None, attempt_num=None, course_id=None,
             previous_question=last_attempt.current_question()
         )
     # allowed to start
-    if not quest_paper.can_attempt_now(user, course_id):
-        msg = "You cannot attempt {0} quiz more than {1} time(s)".format(
-            quest_paper.quiz.description, quest_paper.quiz.attempts_allowed)
+    if not quest_paper.can_attempt_now(user, course_id)[0]:
+        # msg = "You cannot attempt {0} quiz more than {1} time(s)".format(
```
[the remark this answers is not in the document]
Comment (ankitjavalkar) [reply yaksh/views.py]: Will remove this along with addition of test cases

File: yaksh/models.py (hunk 2 above)
Comment (maheshgudi) [reply yaksh/models.py]: Fair enough.

Comment (maheshgudi) [reply yaksh/models.py]: I just believe that a method shouldn't return a message. To me, I don't believe model methods should do that.

File: yaksh/views.py [hunk 4]
```diff
@@ -534,9 +534,8 @@ def start(request, questionpaper_id=None, attempt_num=None, course_id=None,
             previous_question=last_attempt.current_question()
         )
     # allowed to start
-    if not quest_paper.can_attempt_now(user, course_id):
-        msg = "You cannot attempt {0} quiz more than {1} time(s)".format(
-            quest_paper.quiz.description, quest_paper.quiz.attempts_allowed)
+    if not quest_paper.can_attempt_now(user, course_id)[0]:
+        msg = quest_paper.can_attempt_now(user, course_id)[1]
```
Comment (maheshgudi) [inline yaksh/views.py]: We can add the message here. There is not a whole lot of reason for the msg to be generated inside the method

**Pull request merged**